core/src/replay_stage.rs

```diff
@@ -3431,6 +3439,31 @@ impl ReplayStage {
                     ("hash", bank.hash().to_string(), String),
                 );
 
+                // Fast leader handover: if we're going to be the next leader, and our leader window
+                // starts on the next slot, then send the bank through this channel.
+                let is_next_leader = leader_schedule_cache
+                    .slot_leader_at(bank.slot() + 1, Some(bank))
+                    .is_some_and(|leader| &leader == my_pubkey);
+                let next_slot = bank.slot().saturating_add(1);
+
+                if is_next_leader && next_slot == first_of_consecutive_leader_slots(next_slot) {
+                    if let Some(block_id) = block_id {
+                        let start_slot = next_slot;
+                        let end_slot = next_slot.saturating_add(NUM_CONSECUTIVE_LEADER_SLOTS - 1);
+                        let parent_block = (bank.slot(), block_id);
+
+                        let leader_window_info = LeaderWindowInfo {
+                            start_slot,
+                            end_slot,
+                            parent_block,
+                            skip_timer: Instant::now(), // can ignore
+                        };
+
+                        let _ = optimistic_parent_sender
+                            .send_timeout(leader_window_info, Duration::from_secs(1));
```

Comment:
can you explain this one to me?

Seems like we should either just wait for the event to go through, or panic (for testing)/log when we timeout or something